hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeChunkGenerator.java

```diff
@@ -0,0 +1,163 @@
+          "Datanode chunk generator is not supported in secure environment");
+    }
+
+    StorageContainerLocationProtocol scmLocationClient =
+        createStorageContainerLocationClient(ozoneConf);
+    List<Pipeline> pipelines = scmLocationClient.listPipelines();
+
+    Pipeline pipeline;
+    if (pipelineId != null && pipelineId.length() > 0) {
+      pipeline = pipelines.stream()
+          .filter(p -> p.getId().toString().equals(pipelineId))
+          .findFirst()
+          .orElseThrow(() -> new IllegalArgumentException(
+              "Pipeline ID is defined, but there is no such pipeline: "
+                  + pipelineId));
+
+    } else {
+      pipeline = pipelines.stream()
+          .filter(p -> p.getFactor() == ReplicationFactor.THREE)
+          .findFirst()
+          .orElseThrow(() -> new IllegalArgumentException(
+              "Pipeline ID is defined, and no pipeline has been found with "
+                  + "factor=THREE"));
+      LOG.info("Using pipeline {}", pipeline.getId().toString());
+
+    }
+
+    xceiverClientSpi =
+        new XceiverClientManager(ozoneConf).acquireClient(pipeline);
+
+    timer = getMetrics().timer("file-create");
+
+    byte[] data = RandomStringUtils.randomAscii(chunkSize)
+        .getBytes(StandardCharsets.UTF_8);
+
+    dataToWrite = ByteString.copyFrom(data);
+
+    Checksum checksum = new Checksum(ChecksumType.CRC32, chunkSize);
+    checksumProtobuf = checksum.computeChecksum(data).getProtoBufMessage();
+
+    runTests(this::writeChunk);
+
+    return null;
+  }
+
+  private void writeChunk(long stepNo)
+      throws Exception {
+
+    //Always use this fake blockid.
+    DatanodeBlockID blockId = DatanodeBlockID.newBuilder()
+        .setContainerID(1L)
+        .setLocalID(stepNo % 20)
+        .setBlockCommitSequenceId(stepNo)
+        .build();
+
+    ChunkInfo chunkInfo = ChunkInfo.newBuilder()
+        .setChunkName(getPrefix() + "_testdata_chunk_" + stepNo)
+        .setOffset(0)
+        .setLen(chunkSize)
+        .setChecksumData(checksumProtobuf)
+        .build();
+
+    WriteChunkRequestProto.Builder writeChunkRequest =
+        WriteChunkRequestProto
+            .newBuilder()
+            .setBlockID(blockId)
+            .setChunkData(chunkInfo)
+            .setData(dataToWrite);
+
+    String id = xceiverClientSpi.getPipeline().getFirstNode().getUuidString();
+
+    ContainerCommandRequestProto.Builder builder =
+        ContainerCommandRequestProto
+            .newBuilder()
+            .setCmdType(Type.WriteChunk)
+            .setContainerID(blockId.getContainerID())
+            .setDatanodeUuid(id)
+            .setWriteChunk(writeChunkRequest);
+
+    ContainerCommandRequestProto request = builder.build();
+    timer.time(() -> {
+      XceiverClientReply xceiverClientReply =
+          xceiverClientSpi.sendCommandAsync(request);
+      xceiverClientSpi.watchForCommit(xceiverClientReply.getLogIndex(), 1000L);
+      return null;
+    });
```

Comment:
In my tests Freon created fewer chunks than expected (in 10 runs of 100 chunks it created total 711 instead of 1000).  Changing to sync command fixes it:

```
    timer.time(() -> xceiverClientSpi.sendCommand(request));
```

Reply:
That's interesting. Some of the chunks also missing for me. But what can be the problem with the `sendCommandAsync`? Isn't it a ratis bug?

Reply:
> Isn't it a ratis bug?

Might be.  I also tried increasing the timeout of `watchForCommit`, but it didn't help.

Reply:
That's good catch. cc: @mukul1987 @bshashikant